src/test/kotlin/com/pubnub/api/subscribe/SubscribeTest.kt

```diff
@@ -0,0 +1,157 @@
+package com.pubnub.api.subscribe
+
+import com.pubnub.api.managers.EventEngineManager
+import com.pubnub.api.managers.SubscriptionData
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+import io.mockk.CapturingSlot
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.verify
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+private const val CHANNEL_01 = "channel01"
+private const val CHANNEL_02 = "channel02"
+private const val CHANNEL_03 = "channel03"
+
+private const val CHANNEL_GROUPS_01 = "channelGroups01"
+private const val CHANNEL_GROUPS_02 = "channelGroups02"
+private const val CHANNEL_GROUPS_03 = "channelGroups03"
+
+internal class SubscribeTest {
+    private val channelsInSubscriptionData = mutableSetOf(CHANNEL_01, CHANNEL_02)
+    private val channelGroupsInSubscriptionData = mutableSetOf(CHANNEL_GROUPS_01, CHANNEL_GROUPS_02)
+    private lateinit var objectUnderTest: Subscribe
+    private val eventEngineManager: EventEngineManager = mockk()
+    private var subscriptionData: SubscriptionData = createSubscriptionStateContainingValues()
+    private val withPresence = false
+    private val withTimetoken = 12345345452L
+    private val event: CapturingSlot<Event> = slot()
+
+    @BeforeEach
+    internal fun setUp() {
+        every { eventEngineManager.addEventToQueue(capture(event)) } returns Unit
+        objectUnderTest = Subscribe(eventEngineManager, subscriptionData)
+    }
+
+    @Test
+    internal fun `should store channels and channelGroups in local storage and pass SubscriptionChange event for handling when timeToken is zero`() {
+        val channelToSubscribe = CHANNEL_03
+        val channelGroupsToSubscribe = CHANNEL_GROUPS_03
+
+        objectUnderTest.subscribe(listOf(channelToSubscribe), listOf(channelGroupsToSubscribe), withPresence)
+
+        verify { eventEngineManager.addEventToQueue(any()) }
+
+        assertEquals(
+            Event.SubscriptionChanged(
+                (channelsInSubscriptionData + channelToSubscribe).toList(),
+                (channelGroupsInSubscriptionData + channelGroupsToSubscribe).toList()
+            ),
+            event.captured
+        )
+        assertTrue(subscriptionData.channels.contains(CHANNEL_01))
+        assertTrue(subscriptionData.channelGroups.contains(CHANNEL_GROUPS_01))
```

Comment:
What does it actually test here?

Reply:
Maybe it's enough to verify that the we have correct event? I mean, we had to get some of the information from somewhere, so if we're able to create an event with correct content then I think we can omit asserting internals

Reply:
ok. I will modify it.